xdsl/dialects/tensor.py

```diff
@@ -162,10 +173,102 @@ def verify_(self) -> None:
             )
 
 
+@irdl_op_definition
+class ExtractSliceOp(IRDLOperation):
+    name = "tensor.extract_slice"
+
+    source: Operand = operand_def(TensorType)
+    offsets: VarOperand = var_operand_def(IndexType)
+    sizes: VarOperand = var_operand_def(IndexType)
+    strides: VarOperand = var_operand_def(IndexType)
+    static_offsets: DenseArrayBase = prop_def(DenseArrayBase)
+    static_sizes: DenseArrayBase = prop_def(DenseArrayBase)
+    static_strides: DenseArrayBase = prop_def(DenseArrayBase)
+    result: OpResult = result_def(TensorType)
+
+    irdl_options = [AttrSizedOperandSegments(as_property=True)]
+
+    @staticmethod
+    def from_static_parameters(
+        source: SSAValue | Operation,
+        offsets: Sequence[int],
+        sizes: Sequence[int],
+        strides: Sequence[int] | None = None,
+        reduce_rank: bool = False,
+    ) -> ExtractSliceOp:
+
+        if strides is None:
+            strides = [1] * len(offsets)
+        source_v = SSAValue.get(source)
+        source_t = source_v.type
+        if not isinstance(source_t, ContainerType):
+            raise ValueError(f"Expected ContainerType, got {source_t}")
+
+        if reduce_rank:
+            result_sizes = list(s for s in sizes if s != 1)
+        else:
+            result_sizes = list(sizes)
+
+        return_type = TensorType[Any](source_t.get_element_type(), result_sizes)
+
+        return ExtractSliceOp.build(
+            operands=[source, [], [], []],
```

Comment:
This feels a little weird, why not handle these also? Or at least raise NotImplemented if dynamic offsets are passed in

Reply:
This is `from_static_parameters`, it does not take SSAValues arguments for the dynamic offsets,sizes, and strides.

Reply:
The more generic constructor should be easier to adapt than `memref.subview`'s, from which I just ported this one, but I'd be glad to worry about it when someone needs it; I don't as I'll use those to lower compile-time sized code.

Reply:
Oh sorry I misread, I thought it was the init, all good.